datalad/core/distributed/push.py

```diff
@@ -235,7 +231,7 @@ def __call__(
                 lgr.info(
                     "No tracked remote for active branch, "
                     "detection of last pushed state not in effect.")
-        elif since:
+        elif since is not None:
             # will blow with ValueError if unusable
             ds_repo.get_hexsha(since)
```

Comment:
indeed it does guard but I think we could improve UX with a more informative message above instead of
```shell
$> datalad push --since=''   
[ERROR  ] Unknown commit identifier:  [gitrepo.py:format_commit:1721] (ValueError) 
```

Reply:
I quite like that uniform treatment, but I thought you might not.  As I said, feel free to drop what you don't want and add whatever extra code you'd like.